iceoryx_posh/test/moduletests/test_popo_toml_gateway_config_parser.cpp

```diff
@@ -43,192 +56,256 @@ class TomlGatewayConfigParserTest : public TestWithParam<ParseErrorInputFile_t>
     iox::roudi::ConfigFilePathString_t m_configFilePath;
 };
 
+class TomlGatewayConfigParserSuiteTest : public TestWithParam<CheckCharactersValidity_t>
+{
+  public:
+    void SetUp(){};
+    void TearDown()
+    {
+        if (std::remove(TestFilePath.c_str()) != 0)
+        {
+            std::cerr << "Failed to remove temporary file '" << TestFilePath
+                      << "'. You'll have to remove it by yourself." << std::endl;
+        }
+    };
+    void CreateTmpTomlFile(std::shared_ptr<cpptoml::table> toml)
+    {
+        std::fstream fs(TestFilePath, std::fstream::out | std::fstream::trunc);
+        if (fs.std::fstream::is_open())
+        {
+            fs << *toml;
+        }
+        else
+        {
+            ASSERT_STREQ("expected open fstream", "fstream not open");
+        }
+        fs.close();
+    }
+};
+
 // ======================================== Tests ======================================== //
-TEST_F(TomlGatewayConfigParserTest, PassesValidationIfValidCharactersUsedInServiceDescription)
+/// we require INSTANTIATE_TEST_CASE since we support gtest 1.8 for our safety targets
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+
+INSTANTIATE_TEST_CASE_P(ValidTest,
+                        TomlGatewayConfigParserSuiteTest,
+                        ::testing::Values(CheckCharactersValidity_t{std::make_tuple("validcharacters", false)},
+                                          CheckCharactersValidity_t{std::make_tuple("UPPERCASECHARACTERS", false)},
+                                          CheckCharactersValidity_t{std::make_tuple("lowercasecharacters", false)},
+                                          CheckCharactersValidity_t{std::make_tuple("Number1234567890", false)},
+                                          CheckCharactersValidity_t{std::make_tuple("Under_score_Characters", false)},
+                                          CheckCharactersValidity_t{std::make_tuple("_BeginsWithUnderscore", false)},
+                                          CheckCharactersValidity_t{std::make_tuple("Hyphen-InService", true)},
+                                          CheckCharactersValidity_t{std::make_tuple("1234567890", true)},
+                                          CheckCharactersValidity_t{std::make_tuple("這場考試_!*#:", true)}));
+
+#pragma GCC diagnostic pop
+
+TEST_P(TomlGatewayConfigParserSuiteTest, CheckCharactersUsedInServiceDescription)
```

Comment:
Could you please add the following test cases:
- Parse a valid config file with the maximum allowed number of configured services (MAX_GATEWAY_SERVICES)
- Parse a valid config file with more than the maximum allowed number of configured services
- Parse a valid config file where the same (service/instance/event) entry triplet is contained multiple times -> gatewayConfig should then have only one entry (every entry should be unique)